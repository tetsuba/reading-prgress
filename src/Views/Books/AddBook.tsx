import Svg from '../../Components/Svg/Svg'
import Button from '../../Components/Button/Button'
import Modal from '../../Components/Modal/Modal'
import { useState } from 'react'
import RegisterBookForm from './RegisterBookForm'

export default function AddBook() {
    const [showModal, setShowModal] = useState(false)
    return (
        <>
            <Button
                className="flex"
                type="button"
                template="tertiary"
                clickHandler={() => setShowModal(true)}
            >
                <span className="md:mr-2">
                    <Svg type="plus" />
                </span>
                <span className="hidden md:block">Add Book</span>
            </Button>
            {showModal && (
                <Modal className="">
                    <RegisterBookForm
                        clickHandlerCancel={() => setShowModal(false)}
                    />
                </Modal>
            )}
        </>
    )
}
