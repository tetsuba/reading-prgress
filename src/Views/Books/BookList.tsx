import { useMutation, useQueryClient } from 'react-query'
import Button from '../../Components/Button/Button'
import { deleteBook } from '../../lib/service'
import { useDispatch, useSelector } from 'react-redux'
import { userIdSelector } from '../../store/user/userSelectors'
import { addBook } from '../../store/book/bookSlice'
import { useNavigate } from 'react-router-dom'
import Modal from '../../Components/Modal/Modal'
import { useState } from 'react'
import Confirmation from '../../Components/Modal/Confirmation'

type BookTypes = {
    story: string
    title: string
    id: number
}

type PropTypes = {
    list: BookTypes[]
}

export default function BookList(props: PropTypes) {
    const navigate = useNavigate()
    const dispatch = useDispatch()
    const [book, setBook] = useState<null | BookTypes>(null)
    const userId = useSelector(userIdSelector)
    const queryClient = useQueryClient()
    const mutation = useMutation(deleteBook, {
        onSuccess: async (data) => {
            queryClient.setQueryData(['books', userId], data)
        }
    })

    return (
        <div data-testid="book-list">
            <div className="rounded-t-lg bg-gray-200 px-4 py-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-6">
                <div className="text-lg font-bold text-gray-900">
                    Book Title
                </div>
            </div>
            {props.list.map((book, i) => (
                <div
                    key={`book-list-${i}`}
                    className={`px-4 py-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-6 ${
                        i & 1 ? 'bg-white' : 'bg-gray-50'
                    }`}
                >
                    <div className="flex items-center text-base font-medium text-gray-800">
                        {book.title}
                    </div>
                    <div className="mt-1 text-sm text-gray-900 sm:mt-0"></div>
                    <div className="mt-1 flex justify-end text-sm text-gray-900 sm:mt-0">
                        <Button
                            dataTestid="book-list-read"
                            template="secondary"
                            clickHandler={() => {
                                dispatch(addBook(book))
                                navigate('/reading')
                            }}
                        >
                            Read
                        </Button>
                        <Button
                            dataTestid="book-list-delete"
                            template="icon"
                            svg="delete"
                            className="ml-2 hover:bg-gray-200 hover:text-gray-900 dark:hover:bg-gray-800 dark:hover:text-white"
                            clickHandler={() => {
                                setBook(book)
                            }}
                        />
                    </div>
                </div>
            ))}
            {book && (
                <Modal>
                    <Confirmation
                        bookTile={book.title}
                        clickHandlerCancel={() => setBook(null)}
                        clickHandlerDelete={() => {
                            mutation.mutate(`?bookId=${book.id}`)
                            setBook(null)
                        }}
                    />
                </Modal>
            )}
        </div>
    )
}
