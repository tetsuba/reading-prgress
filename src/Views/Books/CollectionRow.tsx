import { ApiCollectionTypes } from '../../lib/service-types'
import { useDispatch } from 'react-redux'
import Row from '../../Components/Row/Row'
import { ViewBooksButton } from '../../Components/Button/Buttons'
import { updateViewBookCollection } from '../../store/view/viewSlice'

type CollectionPropTypes = {
    data?: ApiCollectionTypes | undefined
    index?: number
}

export default function CollectionRow(props: CollectionPropTypes) {
    const dispatch = useDispatch()
    const { data, index = 0 } = props
    const text = props.data
        ? `${props.data.title} (${props.data.books.length})`
        : ''
    return (
        <Row index={index} text={text} iconColor="" icon="library">
            <ViewBooksButton
                onClick={() => data && dispatch(updateViewBookCollection(data))}
            />
        </Row>
    )
}
