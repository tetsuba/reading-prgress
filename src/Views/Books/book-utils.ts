export function getBookStatusColour(history: string | null) {
    if (history) {
        const data = typeof history === 'string' ? JSON.parse(history) : history

        const words = data[data.length - 1].words
        if (words.length) {
            return `text-red-500`
        }
        return 'text-green-500'
    }
    return 'text-gray-300'
}
