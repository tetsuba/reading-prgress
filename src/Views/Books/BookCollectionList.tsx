import Svg from '../../Components/Svg/Svg'
import Button from '../../Components/Button/Button'
import BookList, { BookTypes } from './BookList'
import { useEffect, useState } from 'react'

type CollectionTypes = {
    id: string
    title: string
    description: string
    books: BookTypes[]
}

type PropTypes = {
    collections: CollectionTypes[]
}

export default function BookCollectionList(props: PropTypes) {
    const [showCollections, setShowCollections] = useState(true)
    const [bookCollection, setBookCollection] =
        useState<CollectionTypes | null>(null)

    useEffect(() => {
        if (bookCollection) {
            props.collections.forEach((collection) => {
                if (collection.id === bookCollection.id) {
                    setBookCollection(collection)
                }
            })
        }
    }, [props.collections])

    return (
        <div data-testid="collection-list">
            {showCollections &&
                props.collections.map((collection, i: number) => (
                    <div
                        key={`book-list-${i}`}
                        className={`px-4 py-5 sm:grid sm:grid-cols-2 sm:gap-4 sm:px-6 ${
                            i & 1 ? 'bg-white' : 'bg-gray-50'
                        }`}
                    >
                        <div className="flex items-center text-base font-medium text-gray-800">
                            <span className={`mr-6`}>
                                <Svg type="library" />
                            </span>{' '}
                            {collection.title}
                        </div>
                        <div className="mt-1 flex justify-end text-sm text-gray-900 sm:mt-0">
                            <Button
                                className="flex"
                                dataTestid="collection-button"
                                template="secondary"
                                clickHandler={() => {
                                    setShowCollections(false)
                                    setBookCollection(collection)
                                }}
                            >
                                View Books
                                <span className={`ml-2`}>
                                    <Svg type="eye" />
                                </span>
                            </Button>
                        </div>
                    </div>
                ))}

            {!showCollections && bookCollection && (
                <BookList
                    libId={bookCollection.id}
                    title={bookCollection.title}
                    list={bookCollection.books}
                    clickHandlerBack={() => setShowCollections(true)}
                    delete={bookCollection.id === '001'}
                />
            )}
        </div>
    )
}
