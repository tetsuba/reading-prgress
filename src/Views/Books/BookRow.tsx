import { useDispatch } from 'react-redux'
import { useNavigate } from 'react-router-dom'
import { getIconColorForBookRow } from './book-utils'
import Row from '../../Components/Row/Row'
import {
    DeleteBookButton,
    ReadBookButton
} from '../../Components/Button/Buttons'
import { addBook } from '../../store/book/bookSlice'
import Display from '../../Components/Dispay/Display'
import { ApiBookTypes } from '../../api/api-types'

type BookPropTypes = {
    data?: ApiBookTypes
    index?: number
    collectionId?: string
    deleteBook?: (book: ApiBookTypes) => void
}

export default function BookRow(props: BookPropTypes) {
    const { data, index, collectionId, deleteBook } = props
    const dispatch = useDispatch()
    const navigate = useNavigate()
    const iconColor = getIconColorForBookRow(data?.history)
    return (
        <Row
            index={index || 0}
            text={data?.title}
            icon="bookmark"
            iconColor={iconColor}
        >
            <ReadBookButton
                onClick={() => {
                    if (data && collectionId) {
                        dispatch(addBook({ book: data, libId: collectionId }))
                        navigate('/reading')
                    }
                }}
            />
            <Display value={props.collectionId === '001'}>
                <DeleteBookButton
                    onClick={() => data && deleteBook && deleteBook(data)}
                />
            </Display>
        </Row>
    )
}
