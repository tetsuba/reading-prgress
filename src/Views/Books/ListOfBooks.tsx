import * as R from 'ramda'
import { useDispatch } from 'react-redux'
import { useState, useMemo } from 'react'

// UTILS
import { filterBooksByTitle, getBooks } from './book-utils'

// STORE
import { updateCurrentCollectionId } from '../../store/current/currentSlice'

// COMPONENTS
import Svg from '../../Components/Svg/Svg'
import Input from '../../Components/Form/Input'
import Loop from '../../Components/Loop/Loop'
import BookRow from '../../Components/Row/BookRow'
import Button from '../../Components/Button/Button'

// TYPES
import { CollectionWithBooksIconTypes } from '../../store/selector.types'

type PropTypes = {
    collection: CollectionWithBooksIconTypes | undefined
}

export default function ListOfBooks(props: PropTypes) {
    const dispatch = useDispatch()
    const [search, setSearch] = useState('')
    const books = useMemo(() => getBooks(props), [props.collection])

    if (R.isNil(props.collection)) return <>loading...</>
    const filteredBooks = filterBooksByTitle(books, search)

    return (
        <div data-testid="book-list">
            <div className="flex justify-end px-4 sm:px-6">
                <Button
                    className="mb-6 flex"
                    data-testid="back-button"
                    icon="back"
                    template="primary"
                    onClick={() => dispatch(updateCurrentCollectionId(null))}
                >
                    <span className="ml-2">Back</span>
                </Button>
            </div>
            <div className="flex justify-between bg-gray-200 px-4 py-5 sm:gap-4 sm:px-6 md:rounded-t-lg">
                <div className="flex items-center">
                    <span className="mr-2 md:mr-6">
                        <Svg icon="library" />
                    </span>
                    <span className="hidden text-lg font-bold text-gray-900 md:block">
                        {props.collection.title}
                    </span>
                </div>
                <div className="grow">
                    <Input
                        template="text"
                        data-testid="search"
                        value={search}
                        onChange={(e) => setSearch(e.target.value)}
                        type="text"
                        placeholder="Search"
                    />
                </div>
            </div>
            <Loop array={filteredBooks}>
                <BookRow />
            </Loop>
        </div>
    )
}
