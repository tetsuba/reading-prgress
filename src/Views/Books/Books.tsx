import { useQuery } from 'react-query'
import { useDispatch, useSelector } from 'react-redux'
import * as R from 'ramda'
import { userIdSelector } from '../../store/user/userSelectors'
import { viewBooksCollectionSelector } from '../../store/view/viewSelectors'
import { getBooks } from '../../api/book'

// COMPONENTS
import Header from '../../Components/Header/Header'
import Loading from '../../Components/Loading/Loading'
import Main from '../../Components/Main/Main'
import Display from '../../Components/Dispay/Display'
import ListOfCollections from './ListOfCollections'

import ListOfBooks from './ListOfBooks'
import { updateViewBookCollection } from '../../store/view/viewSlice'
import { useEffect } from 'react'

export default function Books() {
    const dispatch = useDispatch()
    const collection = useSelector(viewBooksCollectionSelector)
    const userId = useSelector(userIdSelector)
    const { data, isSuccess, isLoading } = useQuery(['books', userId], getBooks)

    useEffect(() => {
        if (collection) {
            data?.data.forEach((c) => {
                if (collection && c.id === collection.id) {
                    dispatch(updateViewBookCollection(c))
                }
            })
        }
    }, [data?.data])

    if (isLoading) {
        return <Loading />
    }

    return (
        <>
            <Header text="Books" />
            <Main>
                <Display value={isSuccess && R.isNil(collection)}>
                    <ListOfCollections collections={data?.data} />
                </Display>
                <Display value={R.not(R.isNil(collection))}>
                    <ListOfBooks collection={collection} />
                </Display>
            </Main>
        </>
    )
}
