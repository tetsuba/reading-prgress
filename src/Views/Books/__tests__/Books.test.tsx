import {
    render,
    waitFor,
    screen,
    fireEvent,
    waitForElementToBeRemoved
} from '@testing-library/react'
import { WrapperWith_Store_Query_Router } from '../../../vitest-setup'
import Books from '../Books'
import axios from 'axios'
import { delay } from '../../../lib/utils'
vi.mock('axios')

const mockNavigate = vi.fn()
vi.mock('react-router-dom', () => ({
    ...(vi.importActual('react-router-dom') as object),
    useNavigate: () => mockNavigate
}))

const booksMock = [
    {
        title: 'titleBook 1',
        story: 'story 1',
        id: 1,
        history: '[{"date":"16/02/2023","words":[]}]'
    },
    {
        title: 'titleBook 2',
        story: 'story 2',
        id: 2,
        history: '[{"date":"16/02/2023","words":["this"]}]'
    }
]

const mockData = {
    data: [
        {
            id: '001',
            title: 'title collection',
            description: 'description',
            books: booksMock
        },
        {
            id: '002',
            title: 'title collection',
            description: 'description',
            books: booksMock
        }
    ]
}

const mockDataNewBook = {
    data: [
        {
            id: '001',
            title: 'title',
            description: 'description',
            books: [
                ...booksMock,
                {
                    title: 'titleBook 3',
                    story: 'story 3',
                    id: 2,
                    history: null
                }
            ]
        },
        {
            id: '002',
            title: 'title',
            description: 'description',
            books: booksMock
        }
    ]
}

const mockEventTarget = {
    target: {
        title: { value: 'title3' },
        story: { value: 'story3' },
        difficulty: { value: '3' }
    }
}

describe('Books View', () => {
    test('will render a list of collections', async () => {
        // @ts-ignore
        axios.get.mockResolvedValue(mockData)
        const { asFragment } = render(
            <WrapperWith_Store_Query_Router pathname={'/books'}>
                <Books />
            </WrapperWith_Store_Query_Router>
        )
        await waitFor(() => expect(screen.getByTestId('collection-list')))
        expect(asFragment()).toMatchSnapshot()
    })
    test('opening and closing a collection', async () => {
        // @ts-ignore
        axios.get.mockResolvedValue(mockData)
        const { asFragment } = render(
            <WrapperWith_Store_Query_Router pathname={'/books'}>
                <Books />
            </WrapperWith_Store_Query_Router>
        )
        await waitFor(() => expect(screen.getByTestId('collection-list')))
        fireEvent.click(screen.getAllByTestId('collection-button')[0])
        expect(screen.queryByTestId('book-list')).not.toBeNull()
        fireEvent.click(screen.getByTestId('back-button'))
        expect(screen.queryByTestId('book-list')).toBeNull()
    })
    test('adding a new book', async () => {
        // @ts-ignore
        axios.get.mockResolvedValueOnce(mockData)
        // @ts-ignore
        axios.post.mockResolvedValueOnce(mockDataNewBook)
        const { asFragment } = render(
            <WrapperWith_Store_Query_Router pathname={'/books'}>
                <Books />
            </WrapperWith_Store_Query_Router>
        )
        await waitFor(() => expect(screen.getByTestId('collection-list')))
        fireEvent.click(screen.getAllByTestId('collection-button')[0])
        fireEvent.click(screen.getByText('Add Book'))
        expect(screen.getByTestId('register-book-form'))
        fireEvent.submit(
            screen.getByTestId('register-book-form'),
            mockEventTarget
        )
        await waitForElementToBeRemoved(() =>
            expect(screen.getByTestId('register-book-form'))
        )
        await waitFor(() =>
            expect(screen.getAllByText(/titleBook/)).toHaveLength(3)
        )
    })
    test('deleting a book', async () => {
        // @ts-ignore
        axios.get.mockResolvedValueOnce(mockDataNewBook)
        // @ts-ignore
        axios.delete.mockResolvedValueOnce(mockData)
        const { asFragment } = render(
            <WrapperWith_Store_Query_Router pathname={'/books'}>
                <Books />
            </WrapperWith_Store_Query_Router>
        )
        await waitFor(() => expect(screen.getByTestId('collection-list')))
        fireEvent.click(screen.getAllByTestId('collection-button')[0])
        fireEvent.click(screen.getAllByTestId('book-list-delete')[0])
        expect(screen.getByTestId('modal-confirmation'))
        fireEvent.click(screen.getByTestId('cancel-button'))
        fireEvent.click(screen.getAllByTestId('book-list-delete')[0])
        fireEvent.click(screen.getByTestId('delete-button'))
        await waitFor(() =>
            expect(screen.getAllByText(/titleBook/)).toHaveLength(2)
        )
    })
    test('clicking on read a book', async () => {
        // @ts-ignore
        axios.get.mockResolvedValueOnce(mockData)
        const { asFragment } = render(
            <WrapperWith_Store_Query_Router pathname={'/books'}>
                <Books />
            </WrapperWith_Store_Query_Router>
        )
        await waitFor(() => expect(screen.getByTestId('collection-list')))
        fireEvent.click(screen.getAllByTestId('collection-button')[0])
        fireEvent.click(screen.getAllByTestId('book-list-read')[0])
        await waitFor(() => expect(mockNavigate).toHaveBeenCalled())
    })
    test('will search for a book title', async () => {
        // @ts-ignore
        axios.get.mockResolvedValueOnce(mockData)
        const { asFragment } = render(
            <WrapperWith_Store_Query_Router pathname={'/books'}>
                <Books />
            </WrapperWith_Store_Query_Router>
        )
        await waitFor(() => expect(screen.getByTestId('collection-list')))
        fireEvent.click(screen.getAllByTestId('collection-button')[0])
        fireEvent.change(screen.getByTestId('search'), {
            target: { value: 'titleBook 2' }
        })
        expect(screen.getAllByText(/titleBook/)).toHaveLength(1)
    })
})
