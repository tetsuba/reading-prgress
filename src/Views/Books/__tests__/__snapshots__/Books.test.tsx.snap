// Vitest Snapshot v1

exports[`Books View > will render a list of collections 1`] = `
<DocumentFragment>
  <div>
    <header
      class="bg-white shadow"
    >
      <div
        class="mx-auto max-w-7xl py-6 px-4 sm:px-6 lg:px-8"
      >
        <h1
          class="text-3xl font-bold tracking-tight text-gray-900"
        >
          Books
        </h1>
      </div>
    </header>
    <main>
      <div
        class="mx-auto max-w-7xl py-6 sm:px-6 lg:px-8"
      >
        <div
          class="px-4 py-6 sm:px-0"
        >
          <div
            data-testid="collection-list"
          >
            <div
              class="px-4 py-5 sm:grid sm:grid-cols-2 sm:gap-4 sm:px-6 bg-gray-50"
            >
              <div
                class="flex items-center text-base font-medium text-gray-800"
              >
                <span
                  class="mr-6"
                >
                  <svg
                    class="h-6 w-6"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 21v-8.25M15.75 21v-8.25M8.25 21v-8.25M3 9l9-6 9 6m-1.5 12V10.332A48.36 48.36 0 0012 9.75c-2.551 0-5.056.2-7.5.582V21M3 21h18M12 6.75h.008v.008H12V6.75z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </span>
                 title collection
              </div>
              <div
                class="mt-1 flex justify-end text-sm text-gray-900 sm:mt-0"
              >
                <button
                  class="rounded-lg font-medium text-base text-center px-5 py-2.5 text-gray bg-gray-200 hover:bg-gray-300 flex"
                  data-testid="collection-button"
                >
                  View Books
                  <span
                    class="ml-2"
                  >
                    <svg
                      class="h-6 w-6"
                      fill="none"
                      stroke="currentColor"
                      stroke-width="1.5"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M2.036 12.322a1.012 1.012 0 010-.639C3.423 7.51 7.36 4.5 12 4.5c4.638 0 8.573 3.007 9.963 7.178.07.207.07.431 0 .639C20.577 16.49 16.64 19.5 12 19.5c-4.638 0-8.573-3.007-9.963-7.178z"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                      <path
                        d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <div
              class="px-4 py-5 sm:grid sm:grid-cols-2 sm:gap-4 sm:px-6 bg-white"
            >
              <div
                class="flex items-center text-base font-medium text-gray-800"
              >
                <span
                  class="mr-6"
                >
                  <svg
                    class="h-6 w-6"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 21v-8.25M15.75 21v-8.25M8.25 21v-8.25M3 9l9-6 9 6m-1.5 12V10.332A48.36 48.36 0 0012 9.75c-2.551 0-5.056.2-7.5.582V21M3 21h18M12 6.75h.008v.008H12V6.75z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </span>
                 title collection
              </div>
              <div
                class="mt-1 flex justify-end text-sm text-gray-900 sm:mt-0"
              >
                <button
                  class="rounded-lg font-medium text-base text-center px-5 py-2.5 text-gray bg-gray-200 hover:bg-gray-300 flex"
                  data-testid="collection-button"
                >
                  View Books
                  <span
                    class="ml-2"
                  >
                    <svg
                      class="h-6 w-6"
                      fill="none"
                      stroke="currentColor"
                      stroke-width="1.5"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M2.036 12.322a1.012 1.012 0 010-.639C3.423 7.51 7.36 4.5 12 4.5c4.638 0 8.573 3.007 9.963 7.178.07.207.07.431 0 .639C20.577 16.49 16.64 19.5 12 19.5c-4.638 0-8.573-3.007-9.963-7.178z"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                      <path
                        d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
